Documentation/learning/why.md

```diff
@@ -0,0 +1,14 @@
+# Why etcd
+
+etcd is designed to store a small amount of metadata in consistent, fault-tolerant way. Distributed systems use etcd as the root of scheduling and service discovery configuration storage. [Distributed lock][etcd-concurrency] is implemented on top of etcd, useful for master election in other distributed databases.
+
+Example use cases:
+
+- container linux by CoreOS: Application running on [container linux][container-linux] gets automatic, zero-downtime Linux kernel updates. CoreOS uses [locksmith] to coordinate updates. locksmith uses etcd to implement distributed semaphore, ensuring that only subset of cluster are rebooting at any given time.
+- [Kubernetes][kubernetes] uses etcd for configuration storage for service discovery and cluster management. Kubernetes API server writes cluster states in etcd as persistent storage. And watch API to monitor, roll out critical configuration changes. Consistency is the key to ensure that services correctly schedule and operate.
```

Comment:
[Kubernetes][kubernetes] stores configuration data into etcd for service discovery and cluster management;  etcd's consistency is crucial for correctly scheduling and operating services. The Kubernetes API server persists cluster state into etcd. It uses etcd's watch API to monitor the cluster and roll out critical configuration changes.